File: LC/lc101/lc101.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {boolean}
 */
var isSymmetric = function(root) {
  var helper = function (a, b){
      if (!a) return !b;
      else if (!b) return false;
      else return (a.val == b.val) && helper(a.left, b.right) && helper(a.right, b.left);
  };
  return helper(root.left, root.right);
};

File: LC/lc101/lc101.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def isSymmetric(self, root: TreeNode) -> bool:
        def isMirror(tree1, tree2):
            if not (tree1 or tree2):
                return True
            if not (tree1 and tree2):
                return False
            if tree1.val != tree2.val:
                return False
            return isMirror(tree1.left, tree2.right) and isMirror(tree1.right, tree2.left)
        if not root:
            return True
        return isMirror(root.left, root.right)
        